src/algebra/regular/smul.lean

```diff
@@ -76,35 +77,22 @@ lemma is_left_regular [has_mul R] {a : R} (h : is_smul_regular R a) :
 lemma is_right_regular [has_mul R] {a : R} (h : is_smul_regular R (mul_opposite.op a)) :
   is_right_regular a := h
 
-end has_scalar
-
-section monoid
-
-variables [monoid R] [mul_action R M]
-
-variable (M)
-
-/-- One is `M`-regular always. -/
-@[simp] lemma one : is_smul_regular M (1 : R) :=
-λ a b ab, by rwa [one_smul, one_smul] at ab
-
-variable {M}
-
-lemma mul (ra : is_smul_regular M a) (rb : is_smul_regular M b) :
-  is_smul_regular M (a * b) :=
+lemma mul [has_mul R] [is_scalar_tower R R M]
+  (ra : is_smul_regular M a) (rb : is_smul_regular M b) : is_smul_regular M (a * b) :=
 ra.smul rb
 
-lemma of_mul (ab : is_smul_regular M (a * b)) :
+lemma of_mul [has_mul R] [is_scalar_tower R R M] (ab : is_smul_regular M (a * b)) :
```

Comment:
Huh, I guess `is_scalar_tower R R M` is a super wacky way to say roughly "`mul_action` but without associativity"

Reply:
I thought this was the same instance that was used as the `algebra` alternative, but that's `R M M` instead. Interesting!